pkg/fs/fs_test.go

```diff
@@ -253,6 +253,44 @@ func TestReadFileFromRoot(t *testing.T) {
 	})
 }
 
+func TestTruncateFile(t *testing.T) {
+	condSkip(t)
+	inEmptyMutDir(t, func(env *mountEnv, rootDir string) {
+		tmpFile, err := ioutil.TempFile(rootDir, "camlitest")
+		if err != nil {
+			t.Fatal(err)
+		}
+		if _, err = tmpFile.WriteString("hello world from test"); err != nil {
+			t.Fatal(err)
+		}
+		if err = tmpFile.Close(); err != nil {
+			t.Fatal(err)
+		}
+
+		tmpFile, err = os.OpenFile(tmpFile.Name(), os.O_RDWR, 0644)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if err = tmpFile.Truncate(6); err != nil {
+			t.Fatal(err)
+		}
+		if _, err = tmpFile.WriteAt([]byte("perkeep"), 6); err != nil {
```

Comment:
how about a `tmpFile.Stat` here first and make sure the size is 6 first?

at that point there are 3 uses of '6' so maybe `const truncPos = 6` locally

Reply:
good point, done.